zebra-consensus/src/transaction/tests.rs

```diff
@@ -1170,52 +1175,6 @@ async fn v5_transaction_with_exceeding_expiry_height() {
     );
 }
 
-/// Tests if a coinbase V5 transaction with an expiry height exceeding the
-/// maximum is rejected.
-#[tokio::test]
-async fn v5_coinbase_transaction_with_exceeding_expiry_height() {
-    let state_service =
-        service_fn(|_| async { unreachable!("State service should not be called") });
-    let verifier = Verifier::new(Network::Mainnet, state_service);
-
-    let block_height = block::Height::MAX;
-
-    let (input, output) = mock_coinbase_transparent_output(block_height);
-
-    // This expiry height exceeds the maximum defined by the specification.
-    let expiry_height = block::Height(500_000_000);
-
-    // Create a coinbase V4 tx.
-    let transaction = Transaction::V5 {
-        inputs: vec![input],
-        outputs: vec![output],
-        lock_time: LockTime::unlocked(),
-        expiry_height,
-        sapling_shielded_data: None,
-        orchard_shielded_data: None,
-        network_upgrade: NetworkUpgrade::Nu5,
-    };
-
-    let result = verifier
-        .oneshot(Request::Block {
-            transaction: Arc::new(transaction.clone()),
-            known_utxos: Arc::new(HashMap::new()),
-            height: block_height,
-            time: chrono::MAX_DATETIME,
-        })
-        .await;
-
-    assert_eq!(
-        result,
-        Err(TransactionError::MaximumExpiryHeight {
-            expiry_height,
-            is_coinbase: true,
-            block_height,
-            transaction_hash: transaction.hash(),
-        })
-    );
-}
-
```

Comment:
I deleted this because the rule does not apply NU5-onward

> [NU5 onward] nExpiryHeight MUST be less than or equal to 499999999 for non-coinbase transactions.

and V5 transactions will always be NU5-onward. (And the non-coinbase case is covered by another test in the file.)

Reply:
Looks good to me, leaving the comment open so someone else can double-check.

Reply:
I think this is covered by the other tests, at least against Testnet heights, do we need to test that any u32 -1 height is a valid v5 coinbase expiry, with some mock block?

Reply:
@dconnolly yep, the other rule is tested in `v5_coinbase_transaction_expiry_height`.

Good call, I added that test (make sure coinbase is not subject to the non-coinbase bound) in 6b582a871

Reply:
🎉